pkg/splunk/reconcile/util.go

```diff
@@ -204,3 +204,20 @@ func MergePodSpecUpdates(current *corev1.PodSpec, revised *corev1.PodSpec, name
 
 	return result
 }
+
+// MergeServiceSpecUpdates merges the current and revised spec of the service object
+func MergeServiceSpecUpdates(current *corev1.ServiceSpec, revised *corev1.ServiceSpec, name string) bool {
+	scopedLog := log.WithName("MergeServiceSpecUpdates").WithValues("name", name)
+	result := false
+
+	// check for changes in Ports
```

Comment:
Seems like we should be checking more than ports.. If a user modifies an IndexerCluster CR for example to include a serviceTemplate that previously was nil, there could be many additional changes beyond just ports. For example, type may change from ClusterIP to NodePort.

Reply:
Let me look closer here.

Reply:
Added more checks for changes in service spec specifically on ports, external ips, type, external name.